screens/RegisterScreen.tsx

```diff
@@ -1,42 +1,45 @@
 import { NativeStackScreenProps } from '@react-navigation/native-stack';
 import { useState } from 'react';
-import { Alert, Image, ScrollView, TouchableOpacity, View } from 'react-native';
+import { Image, ScrollView, TouchableOpacity, View } from 'react-native';
 import { Button, Text, TextInput, useTheme } from 'react-native-paper';
 import { SafeAreaView } from 'react-native-safe-area-context';
 import hushallet_logo from '../assets/logo/hushallet_logo.png';
 import { RootStackParamList } from '../navigators/RootStackNavigator';
-import { authStyles } from './LoginScreen';
+import {
+  createUser,
+  resetState,
+  selectLogInSuccess,
+} from '../store/Auth/slice';
+import { useAppDispatch, useAppSelector } from '../store/store';
+import { authStyles } from '../themes/styles';
 
 type Props = NativeStackScreenProps<RootStackParamList, 'Register'>;
 
 const RegisterScreen = ({ navigation }: Props) => {
   const { colors } = useTheme();
   const [form, setForm] = useState({ username: '', password: '' });
-
-  //const [isSubmitting, setIsSubmitting] = useState(false);
-
   const [showPassword, setShowPassword] = useState(false);
 
-  const submit = async () => {
-    if (!form.password || !form.username) {
-      Alert.alert('Error', 'Please fill in all fields');
-    } else {
-      navigation.replace('HomeNavigator');
+  const dispatch = useAppDispatch();
+  const error = useAppSelector((state) => state.auth.error);
+  const loading = useAppSelector((state) => state.auth.loading);
+  const success = useAppSelector(selectLogInSuccess);
+
+  const handleRegister = async () => {
+    await dispatch(
+      createUser({ username: form.username, password: form.password }),
+    );
+    if (success) {
+      console.log('User created');
+      navigation.navigate('HomeNavigator');
```

Comment:
Replace so that successful sign up means you get redirected to HomeNavigator without the ability to go back to this screen when logged in
```suggestion
      navigation.replace('HomeNavigator');
```